### PR TITLE
replace useGetPaymentRequestIdsQuery by useMyRewardsInfiniteList

### PR DESCRIPTION
**Pull request by pixelfact**

### CODE REVIEW

File: src/api/me/queries.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { components } from "src/__generated/api";
+import { useInfiniteBaseQuery } from "../useInfiniteBaseQuery";
+import { API_PATH } from "../ApiPath";
+import { ME_TAGS } from "./tags";
+
+export type useMyRewardsInfiniteListResponse = components["schemas"]["MyRewardsPageResponse"];
```
Comment (haydencleary) [inline src/api/me/queries.ts:6]:
```suggestion
export type UseMyRewardsInfiniteListResponse = components["schemas"]["MyRewardsPageResponse"];
```

File: src/api/me/queries.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+const useMyRewardsInfiniteList = (
+  params: Partial<Parameters<typeof useInfiniteBaseQuery>[0]>,
```
Comment (haydencleary) [inline src/api/me/queries.ts:9]: Missing generic on `typeof useInfiniteBaseQuery`

File: src/App/Layout/Header/ProfileButton/ViewMobile.tsx [hunk 3]
```diff
@@ -43,11 +45,18 @@ export default function ViewMobile({
...
+    isAscending: false,
+    storageKey: "myRewardsSorting",
   });
-  const hasRewards = paymentRequestIdsQueryData?.githubUsersByPk?.paymentRequests.length || 0 > 0;
+
+  const { data, isLoading, isError } = MeApi.queries.useMyRewardsInfiniteList({
+    queryParams,
+  });
+
+  const rewards = data?.pages.flatMap(page => page.rewards) || [];
```
Comment (haydencleary) [inline src/App/Layout/Header/ProfileButton/ViewMobile.tsx:58]:
```suggestion
  const rewards = data?.pages.flatMap(({rewards}) => rewards) ?? [];
```

File: src/App/Layout/Header/ProfileButton/ViewMobile.tsx [hunk 4]
```diff
@@ -43,11 +45,18 @@ export default function ViewMobile({
...
+  const hasRewards = rewards.length > 0 && !isLoading && !isError;
```
Comment (haydencleary) [inline src/App/Layout/Header/ProfileButton/ViewMobile.tsx:59]:
```suggestion
  const hasRewards = rewards.length && !isLoading && !isError;
```

File: src/App/Layout/Header/index.tsx [hunk 5]
```diff
@@ -18,16 +21,22 @@ export default function Header() {
...
+    field: Fields.Date,
+    isAscending: false,
+    storageKey: "myRewardsSorting",
   });
 
+  const { data, isLoading, isError } = MeApi.queries.useMyRewardsInfiniteList({
+    queryParams,
+  });
+
+  const rewards = data?.pages.flatMap(page => page.rewards) || [];
```
Comment (haydencleary) [inline src/App/Layout/Header/index.tsx:34]:
```suggestion
  const rewards = data?.pages.flatMap(({rewards})=> rewards) ?? [];
```

File: src/App/Layout/Header/index.tsx [hunk 6]
```diff
@@ -18,16 +21,22 @@ export default function Header() {
...
+  const hasRewards = rewards.length > 0 && !isLoading && !isError;
```
Comment (haydencleary) [inline src/App/Layout/Header/index.tsx:35]:
```suggestion
  const hasRewards = rewards.length && !isLoading && !isError;
```

File: src/api/me/queries.ts (hunk 2 above)
Comment (pixelfact) [reply src/api/me/queries.ts]: fixed

File: src/components/UserRewardTable/DesktopUserRewardList.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+  if (error) {
+    return (
+      <div className="mx-auto flex max-w-7xl flex-col gap-6 px-4 py-4 xl:p-8">
+        <ErrorFallback />
+      </div>
+    );
+  }
+
+  const rewards = data?.pages.flatMap(page => page.rewards) || [];
```
Comment (pixelfact) [inline src/components/UserRewardTable/DesktopUserRewardList.tsx:38]:
```suggestion
  const rewards = data?.pages.flatMap(({ rewards }) => rewards) ?? [];
```

File: src/components/UserRewardTable/MobileUserRewardList.tsx [hunk 8]
```diff
@@ -13,55 +13,77 @@ import { ReactNode } from "react";
...
+
+  if (error) {
+    return (
+      <div className="mx-auto flex max-w-7xl flex-col gap-6 px-4 py-4 xl:p-8">
+        <ErrorFallback />
+      </div>
+    );
+  }
+
+  const rewards = data?.pages.flatMap(page => page.rewards) || [];
```
Comment (pixelfact) [inline src/components/UserRewardTable/MobileUserRewardList.tsx:45]:
```suggestion
  const rewards = data?.pages.flatMap(({ rewards }) => rewards) ?? [];
```

**Pull request merged**